Use something else than <filesystem>

@CoryMartin-NOAA reported the issue. Apparently <filesystem> isn't part of the c++ standard lib on WCOSS. Try to use something older ...

Guilty line:

https://github.com/NOAA-EMC/GDASApp/blob/45c2d168024d80c65ece94c5d85a2c1a6f995fe7/utils/soca/gdas_postprocincr.h#L268

Ask cory martin.  I haven't been tracking it and the current global workflow GDAS build fails much earlier 


 time sh -x build_gdas.sh
+ set -eux
+ OPTIND=1
+ _opts='-f '
+ getopts :j:dv option
+ shift 0
+ BUILD_JOBS=8
+ WORKFLOW_BUILD=ON
+ ./gdas.cd/build.sh -f -f
Building GDASApp on unknown target: acorn
Configuring ...
+ cmake -DCLONE_JCSDADATA=NO -DWORKFLOW_TESTS=ON /u/george.vandenberghe/ns/workflow/w0305/sorc/gdas.cd/bundle
CMake Error at CMakeLists.txt:8 (find_package):
  Could not find a package configuration file provided by "ecbuild"
  (requested version 3.5) with any of the following names:

    ecbuildConfig.cmake
    ecbuild-config.cmake

  Add the installation prefix of "ecbuild" to CMAKE_PREFIX_PATH or set
  "ecbuild_DIR" to a directory containing one of the above files.  If
  "ecbuild" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!

real    0m0.079s
user    0m0.019s
sys     0m0.027s

I think I've replicated it in acorn directory
/lfs/h1/emc/nceplibs/noscrub/gwv/filesystem.h.problem/src
cc -c fils.c will produce
 In file included from fils.c:1:
./gdas_incr_handler.h:3:10: fatal error: 'filesystem' file not found
#include <filesystem>


Copy this src to some writable directory and do that cc command and you'll
get the same behavior I think


On Wed, Mar 6, 2024 at 9:23 PM Cory Martin ***@***.***> wrote:

> @GeorgeVandenberghe-NOAA <https://github.com/GeorgeVandenberghe-NOAA>
> probably. I think we can try creating a very small piece of C++ code that
> uses the filesystem library. Let me and @guillaumevernieres
> <https://github.com/guillaumevernieres> chat tomorrow and get back to you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/GDASApp/issues/812#issuecomment-1981809814>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FRPENUOR4O3FUOA6JDYW6COTAVCNFSM6AAAAABA2GRZHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBRHAYDSOBRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

George W Vandenberghe

*Lynker Technologies at * NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

***@***.***


301-683-3769(work)  3017751547(cell)

--- a/utils/fv3jedi/fv3jedi_fv3inc.h
+++ b/utils/fv3jedi/fv3jedi_fv3inc.h
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <filesystem>
 #include <iostream>
 #include <memory>
 #include <string>

--- a/utils/soca/CMakeLists.txt
+++ b/utils/soca/CMakeLists.txt
@@ -1,14 +1,16 @@
 # Increment post processing
 ecbuild_add_executable( TARGET gdas_incr_handler.x
                         SOURCES gdas_incr_handler.cc gdas_postprocincr.h)
-target_compile_features( gdas_incr_handler.x PUBLIC cxx_std_17)
+target_compile_features( gdas_incr_handler.x PUBLIC cxx_std_11)
 target_link_libraries( gdas_incr_handler.x PUBLIC NetCDF::NetCDF_CXX oops atlas soca)
+target_link_libraries( gdas_incr_handler.x PRIVATE stdc++fs)
 
 # Ensemble members post processing
 ecbuild_add_executable( TARGET gdas_ens_handler.x
                         SOURCES gdas_ens_handler.cc gdas_postprocincr.h)
-target_compile_features( gdas_ens_handler.x PUBLIC cxx_std_17)
+target_compile_features( gdas_ens_handler.x PUBLIC cxx_std_11)
 target_link_libraries( gdas_ens_handler.x PUBLIC NetCDF::NetCDF_CXX oops atlas soca)
+target_link_libraries( gdas_ens_handler.x PRIVATE stdc++fs)
 
 # Hybrid-Weight
 ecbuild_add_executable( TARGET gdas_socahybridweights.x

--- a/utils/soca/gdas_ens_handler.h
+++ b/utils/soca/gdas_ens_handler.h
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <filesystem>
 #include <iostream>
 #include <string>
 #include <vector>

--- a/utils/soca/gdas_incr_handler.h
+++ b/utils/soca/gdas_incr_handler.h
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <filesystem>
 #include <iostream>
 #include <string>
 

--- a/utils/soca/gdas_postprocincr.h
+++ b/utils/soca/gdas_postprocincr.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <filesystem>
+#include <experimental/filesystem>
 
 #include <iostream>
 #include <string>
@@ -260,12 +260,12 @@ class PostProcIncr {
   std::string socaFname() {
     std::string datadir;
     outputIncrConfig_.get("datadir", datadir);
-    std::filesystem::path pathToResolve = datadir;
+    std::experimental::filesystem::path pathToResolve(datadir);
     std::string exp;
     outputIncrConfig_.get("exp", exp);
     std::string outputType;
     outputIncrConfig_.get("type", outputType);
-    std::string incrFname = std::filesystem::canonical(pathToResolve);
+    std::string incrFname = std::experimental::filesystem::canonical(pathToResolve);
     incrFname += "/ocn." + exp + "." + outputType + "." + dt_.toString() + ".nc";
 
     return incrFname;

--- a/utils/soca/gdas_socahybridweights.h
+++ b/utils/soca/gdas_socahybridweights.h
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <filesystem>
 #include <iostream>
 #include <string>
 #include <vector>

